src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -432,4 +443,26 @@ private function loadFromExtensions($content)
             $this->container->loadFromExtension($namespace, $values);
         }
     }
+
+    /**
+     * Checks the keywords used to define a service.
+     *
+     * @param string $id         The service name
+     * @param array  $definition The service definition to check
+     *
+     * @throws InvalidArgumentException When an invalid keyword is used
+     */
+    private static function checkDefinition($id, array $definition)
+    {
+        foreach ($definition as $key => $value) {
+            if (!in_array($key, static::$keywords)) {
+                throw new InvalidArgumentException(sprintf(
```

Comment:
The problem is when you have a generic bundle that supports ``"symfony/symfony": "^2.3|^3.0"``, a solution to support the new api is to have the new and the deprecated keywords in its configuration files.
This PR would force people to update their package in a minor version and it is against the [symfony BC policy](http://symfony.com/doc/current/contributing/code/bc.html). IMO this exception should be added in symfony 4.0 and unsupported keywords deprecated in symfony 3.1.